src/services/RoomService.ts

```diff
@@ -61,6 +63,35 @@ class RoomService implements IService<Room> {
 
   }
 
+  public async disableById(
+    id: number,
+    roomData: RoomDisableBodyValidator,
+  ): Promise<Room> {
+
+    const roomToDisable = await this.getByKey(
```

Comment:
All of this should be part of the room update method, except the disabling bookings logic that would be another responsibility and we can encapsulate it in a function